src/go/vtunnel/pkg/vmsock/conn_linux.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright © 2022 SUSE LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmsock
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/linuxkit/virtsock/pkg/vsock"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/wsl-helper/pkg/dockerproxy/util"
+	"github.com/sirupsen/logrus"
+)
+
+// PeerHanshake listens for incoming VSOCK connections from the Host process
+// The handshake is perfomed once during startup/restart to make sure that
+// host process is talking to a right hyper-v VM (most likely WSL)
+func PeerHandshake() {
+	l, err := vsock.Listen(vsock.CIDAny, PeerHandshakePort)
+	if err != nil {
+		logrus.Fatalf("PeerHandshake listen for incoming vsock: %v", err)
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			logrus.Errorf("PeerHandshake accepting incoming socket connection: %v", err)
+		}
+		_, err = conn.Write([]byte(SeedPhrase))
+		if err != nil {
+			logrus.Errorf("PeerHandshake writing seed phrase: %v", err)
+		}
+
+		conn.Close()
+		logrus.Info("successful handshake with vsock-host")
+	}
+}
+
+func ListenTCP(addr string, port int) error {
+	l, err := net.ListenTCP("tcp", &net.TCPAddr{IP: net.ParseIP(addr), Port: port})
+	if err != nil {
+		return fmt.Errorf("ListenTCP: %w", err)
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			logrus.Errorf("ListenTCP accept connection: %v", err)
```

Comment:
Wouldn't a likely error here be we're trying to quit (and therefore we should stop looping)?

Reply:
I don't think we want to stop looping just because of the error. An error can be caused by a client who is attempting to connect. Therefore, the loop should remain so they can try again. This is a standard TCP loop to accept conn... unless I'm misunderstanding your question.

Reply:
Right, it's only if the error is `ErrClosed` or something? Whatever the result is when you're shutting down the app.